Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -785,6 +783,9 @@ WarpXParticleContainer::GetChargeDensity (int lev, bool local)
 
 #ifdef WARPX_DIM_RZ
     WarpX::GetInstance().ApplyInverseVolumeScalingToChargeDensity(rho.get(), lev);
+#else
+    // Reflect density over PEC boundaries, if needed.
+    WarpX::GetInstance().ApplyRhofieldBoundary(lev, rho.get(), PatchType::fine);
 #endif
 
     if (local == false) { ablastr::utils::communication::SumBoundary(*rho, WarpX::do_single_precision_comms, gm.periodicity()); }
```

Comment:
Something I've seen is that the AMReX routines handling the periodicity will only copy data from valid cells to the periodic guard cells. So, if there is a corner where one side is periodic and one is PEC, if the periodicity is handled after the PEC boundary, the data in the corner guard cells won't be set. So perhaps move this line above the lines setting the PEC boundaries.